sklearn/decomposition/tests/test_factor_analysis.py

```diff
@@ -83,3 +83,29 @@ def test_factor_analysis():
         precision = fa.get_precision()
         assert_array_almost_equal(np.dot(cov, precision),
                                   np.eye(X.shape[1]), 12)
+
+    # test rotation
+    n_comps = 2
+
+    results = []
+    for method in (None, "varimax", 'quartimax'):
+        fa_var = FactorAnalysis(n_components=n_comps, rotation=method)
+        results.append(fa_var.fit_transform(X))
+    for a, b in ((0, 1), (0, 2), (1, 2)):
+        assert not np.allclose(results[a], results[b])
+
+    assert_raises(NotImplementedError,
+                  FactorAnalysis(rotation='not_implemented').fit_transform, X)
+
+    # test against R's psych::principal with rotate="varimax"
+    # R's factor analysis returns quite different values; therefore, we only
+    # test the rotation itself
+    factors = np.array(
+        [[0.89421016, -0.35854928, -0.27770122, 0.03773647],
```

Comment:
Is there anything special about this input? How was it generated? Please leave a comment.